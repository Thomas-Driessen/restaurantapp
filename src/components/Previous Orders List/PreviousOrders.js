import React from 'react';
import Grid from '@material-ui/core/Grid';
import Order from './Order';
import tableId from '../TableId'

class PreviousOrders extends React.Component {
    constructor(){
        super();
        this.state = { previousDrinks: [],
            previousFoods: []
          };
    }
    componentDidMount() {
        let mounted = true;
        fetch(`/api/orderdrink/${tableId}`)
        .then(res => res.json())
        .then((data) => {
            if(mounted){
                this.setState({ previousDrinks: data })
            }
        })
        .catch(console.log)
        fetch(`/api/orderfood/${tableId}`)
        .then(res => res.json())
        .then((data) => {
            if(mounted){
                this.setState({ previousFoods: data })
            }
        })
        .catch(console.log)

        return() => mounted = false;
    }
    render(){
    return(
    <div>
        <h2>These are your previous orders</h2>
    <Grid container spacing={0} style={{padding: 15}}>
        { this.state.previousFoods.map(currentItem => (
            <Grid key={currentItem.id} item xs={12} sm={6} lg={4} xl={3}>
                <Order key={currentItem.id} orderId={currentItem.foodId} productType="food"/>
            </Grid>
        ))}
        { this.state.previousDrinks.map(currentItem => (
            <Grid key={currentItem.id} item xs={12} sm={6} lg={4} xl={3}>
                <Order key={currentItem.id} orderId={currentItem.drinkId} productType="drink"/>
            </Grid>
        ))}
    </Grid>
    </div>
    )}
}
export default PreviousOrders