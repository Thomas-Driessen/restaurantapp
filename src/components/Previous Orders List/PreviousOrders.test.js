import React from 'react';
import { render } from '@testing-library/react';
import PreviousOrders from './PreviousOrders';
import renderer from 'react-test-renderer';
import { createMuiTheme } from '@material-ui/core/styles';

const theme = createMuiTheme({
  palette: {
    primary: {
      light: '#f05545',
      main: '#b71c1c',
      dark: '#7f0000',
      contrastText: '#fff',
    },
    secondary: {
      light: '#ff7961',
      main: '#f44336',
      dark: '#ba000d',
      contrastText: '#000',
    },
  },
});

const previousFoods = [
  {
    id: 1,
    food: "Pizza"
  },
  {
    id: 2,
    food: "Pasta"
  }
];
const previousDrinks = [
  {
    id: 1,
    drink: "Coca-Cola"
  },
  {
    id: 2,
    drink: "Pepsi"
  }
];

it('PreviousOrders render correctly', () => {
  const tree = renderer
    .create(<PreviousOrders theme={theme} previousFoods={previousFoods} previousDrinks={previousDrinks} totalPrice={0} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render PreviousOrders', () => {
  const { getByText } = render(<PreviousOrders theme={theme} previousFoods={previousFoods} previousDrinks={previousDrinks} totalPrice={0} />);
  const linkElement = getByText(/These are your previous orders/i);
  expect(linkElement).toBeInTheDocument();
});

afterAll(() => {
  global.gc && global.gc()
})