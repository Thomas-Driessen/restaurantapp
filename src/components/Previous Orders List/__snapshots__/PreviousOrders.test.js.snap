// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviousOrders render correctly 1`] = `
<div>
  <h2
    style={
      Object {
        "paddingLeft": 20,
        "textDecorationColor": "primary",
        "textDecorationLine": "underline",
      }
    }
  >
    These are your previous orders
  </h2>
  <h3
    style={
      Object {
        "paddingLeft": 20,
      }
    }
  >
    Foods
  </h3>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    style={
      Object {
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 15,
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    >
      <div>
        <div
          className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            className="MuiCardContent-root"
          >
            <h2
              className="MuiTypography-root MuiTypography-gutterBottom"
            >
              <span
                style={
                  Object {
                    "color": "green",
                    "float": "right",
                  }
                }
              >
                €
              </span>
            </h2>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    >
      <div>
        <div
          className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            className="MuiCardContent-root"
          >
            <h2
              className="MuiTypography-root MuiTypography-gutterBottom"
            >
              <span
                style={
                  Object {
                    "color": "green",
                    "float": "right",
                  }
                }
              >
                €
              </span>
            </h2>
          </div>
        </div>
      </div>
    </div>
  </div>
  <h3
    style={
      Object {
        "paddingLeft": 20,
      }
    }
  >
    Drinks
  </h3>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    style={
      Object {
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 15,
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    >
      <div>
        <div
          className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            className="MuiCardContent-root"
          >
            <h2
              className="MuiTypography-root MuiTypography-gutterBottom"
            >
              <span
                style={
                  Object {
                    "color": "green",
                    "float": "right",
                  }
                }
              >
                €
              </span>
            </h2>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    >
      <div>
        <div
          className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            className="MuiCardContent-root"
          >
            <h2
              className="MuiTypography-root MuiTypography-gutterBottom"
            >
              <span
                style={
                  Object {
                    "color": "green",
                    "float": "right",
                  }
                }
              >
                €
              </span>
            </h2>
          </div>
        </div>
      </div>
    </div>
  </div>
  <h3
    style={
      Object {
        "float": "right",
        "paddingRight": 20,
      }
    }
  >
    Total to be paid:  
    <span
      style={
        Object {
          "color": "green",
        }
      }
    >
      0.00
      €
    </span>
  </h3>
</div>
`;
